classes/Cron.php

```diff
@@ -146,27 +146,48 @@ public static function parseWebmention($request)
             // parse the Mf2 array to a comment array
             $result = \IndieWeb\comments\parse($mf2['items'][0], $target, 1000, 20);
 
+            // sometimes, the author name ends up in the url field
+            if (!empty($result['author']['url']) && !Str::isUrl($result['author']['url'])) {
+                if (empty($result['author']['name'])) {
+                    $result['author']['name'] = $result['author']['url'];
+                }
+                unset($result['author']['url']);
+            }
+
             // php-comments does not do rel=author
-            if ($result['author']['url'] === false && array_key_exists('rels', $mf2) && array_key_exists('author', $mf2['rels']) && array_key_exists(0, $mf2['rels']['author']) && is_string($mf2['rels']['author'][0])) {
+            if (array_key_exists('url', $result['author']) && $result['author']['url'] === false && array_key_exists('rels', $mf2) && array_key_exists('author', $mf2['rels']) && array_key_exists(0, $mf2['rels']['author']) && is_string($mf2['rels']['author'][0])) {
                 $result['author']['url'] = $mf2['rels']['author'][0];
             }
 
             // if h-card is not embedded in h-entry, php-comments returns no author; check for h-card in mf2 output and fill in missing
-            // TODO: align with algorithm outlined in https://indieweb.org/authorship
             foreach ($mf2['items'] as $mf2item) {
                 if ($mf2item['type'][0] == 'h-card') {
-                    if ($result['author']['name'] == ''  && isset($mf2item['properties']['name'][0])) {
+                    $hcardfound = true;
+                    if (empty($result['author']['name'])  && !empty($mf2item['properties']['name'][0])) {
                         $result['author']['name'] = $mf2item['properties']['name'][0];
                     }
-                    if ($result['author']['photo'] == '' && isset($mf2item['properties']['photo'][0])) {
+                    if (empty($result['author']['photo']) && !empty($mf2item['properties']['photo'][0])) {
                         $result['author']['photo'] = $mf2item['properties']['photo'][0];
                     }
-                    if ($result['author']['url'] == ''  && isset($mf2item['properties']['url'][0])) {
+                    if (empty($result['author']['url']) && !empty($mf2item['properties']['url'][0])) {
                         $result['author']['url'] = $mf2item['properties']['url'][0];
                     }
                 }
             }
 
+            // if no h-card was found, try to use 'author' property of h-entry instead
+            if (!$hcardfound ?? false) {
```

Comment:
This is quite elegant, but very hard to read … is would be easier to understand, if you set `$hcardfound = false` before the foreach-loop and remove the ` ?? false` part.

Reply:
Agreed. I'll make a mental note and clean this up later - I want to revisit this flow one more time anyways...